### ISSUE TITLE
Error while exporting from keras format to TF SavedModel format, tf 2.16.1 and keras 3.3.3

### ISSUE THREAD

**Issue by inigojpuente** [keras-team/keras#19876]
ValueError: Exception encountered when calling TimeDistributed.call()

Invalid dtype: NoneType
    
    Arguments received by TimeDistributed.call():
      • inputs=tf.Tensor(shape=(None, None, 10, 8), dtype=float32)
      • training=None
      • mask=None

**Comment by mehtamansi29** [comment]
Hi @inigojpuente - 

Could you help me with code snippet to reproduce this issue ?

**Comment by inigojpuente** [comment]
Hey,

Thanks for getting back.

I was able to solve my issue by modifying the following line in the method that creates the sequential model:

`model_lstm.add(keras.layers.InputLayer(shape=(n_steps, n_length, n_features)))`

instead of
`model_lstm.add(keras.layers.InputLayer(shape=(None, n_length, n_features)))`

In case this may be of use to others and to be clear, this only affected the behaviour of `model.export()` to `SavedModel `format. I was able to train, evaluate and save the model to both `h5 `and `keras `formats with the original implementation.
